engagement_database/reset_message_labels.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+import json
+import subprocess
+
+from core_data_modules.cleaners import Codes, URNCleaner
+from core_data_modules.logging import Logger
+from engagement_database import EngagementDatabase
+from engagement_database.data_models import HistoryEntryOrigin, CommandLogEntry, CommandStatuses
+from google.cloud import firestore
+from id_infrastructure.firestore_uuid_table import FirestoreUuidTable
+from storage.google_cloud import google_cloud_utils
+
+log = Logger(__name__)
+
+BATCH_SIZE = 500
+
+
+@firestore.transactional
+def reset_message_labels(transaction, engagement_db, message_id, dry_run=False):
+    message = engagement_db.get_message(message_id, transaction)
+    assert message is not None, f"Message '{message_id}' does not exist in database"
+
+    # Reset the message's labels and previous_datasets, and return the message to its original dataset
+    message.labels = []
+    message.dataset = message.previous_datasets[0]
+    message.previous_datasets = []
+
+    if not dry_run:
+        engagement_db.set_message(
+            message,
+            HistoryEntryOrigin("Reset message labels", {}),
+            transaction
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Resets all of a message's labels and previous_datasets, and returns "
+                                                 "the message back to its original dataset")
+
+    parser.add_argument("--dry-run", const=True, default=False, action="store_const")
+    parser.add_argument("user", help="Identifier of the user launching this program")
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("engagement_database_credentials_file_url", metavar="engagement-database-credentials-file-url",
+                        help="GS URL of the credentials for the Firestore project to export")
```

Comment:
`help="GS URL to the Engagement firestore database credentials file")`

Reply:
updated